include/cnl/_impl/rounding/tie_to_pos_inf_rounding_tag.h

```diff
@@ -18,7 +24,64 @@ namespace cnl {
     /// \sa cnl::rounding_integer,
     /// cnl::add, cnl::convert, cnl::divide, cnl::left_shift, cnl::multiply, cnl::subtract,
     /// cnl::tie_to_pos_inf_rounding_tag
-    struct tie_to_pos_inf_rounding_tag {
+    struct tie_to_pos_inf_rounding_tag
+            : _impl::homogeneous_deduction_tag_base, _impl::homogeneous_operator_tag_base {
+    };
+
+    namespace _impl {
+        template<>
+        struct is_rounding_tag<tie_to_pos_inf_rounding_tag> : std::true_type {
+        };
+    }
+
+    template<class Operator, typename Operand>
+    struct unary_operator<Operator, tie_to_pos_inf_rounding_tag, Operand>
+            : unary_operator<Operator, _impl::native_tag, Operand> {
+    };
+
+    template<class Operator, typename Lhs, typename Rhs>
+    struct binary_operator<Operator, tie_to_pos_inf_rounding_tag, tie_to_pos_inf_rounding_tag, Lhs, Rhs>
+            : Operator {
+    };
+
+    template<typename Lhs, typename Rhs>
+    struct binary_operator<_impl::divide_op, tie_to_pos_inf_rounding_tag, tie_to_pos_inf_rounding_tag, Lhs, Rhs> {
+    private:
+        CNL_NODISCARD constexpr auto myabs(Lhs const& lhs) const
```

Comment:
@hbe72 unfortunately, until I switch to C++ 20, it's not really worth ditching C++11 which means trailing return types:

```c++
CNL_NODISCARD constexpr auto myabs(Lhs const& lhs) const
-> decltype(lhs < 0 ? -lhs : lhs)
```

Reply:
@hbe72 I've just pushed a fix for this error and a weirdness with OS X where its int64 is different to Linux'.